File: Back- End/models/TripPlace.js
const mongoose = require('mongoose');

const tripplaceSchema = new mongoose.Schema({

    district: {
        type: String,
        required: true,
    },
    name: {
        type: String,
        required: true,
    },
    location: {
        type: String,
        required: true,
    },
    likes: {
        type: Number,
        default: 0,
    },
    description: {
        type: String,
        required: true,
    },
    images: {
        type: Array,
        required: true,
    },
    weather: {
        type: String,
        required: false,
    },
    budget: {
        type: String,
        required: false,
    },
    tripPersonType: {
        type: String,
        required: false,
    },
    tripType: {
        type: String,
        required: false,
    },
    locationLink: {
        type: String,
        required: false,
    },
});

const TripPlace = mongoose.model('TripPlace', tripplaceSchema);
module.exports = TripPlace;

File: Front- End/lib/Models/TripPlace.dart
class TripPlace {
  final String district;
  final String name;
  final String location;
  final String description;
  final List<String> images;
  final String tripType;
  final String locationLink;

  TripPlace({
    required this.district,
    required this.name,
    required this.location,
    required this.description,
    required this.images,
    required this.tripType,
    required this.locationLink,
  });
    factory TripPlace.fromJson(Map<String, dynamic> json) {
    return TripPlace(
      district: json['district'],
      name: json['name'],
      location: json['location'],
      description: json['description'],
      images: List<String>.from(json['images']),
      tripType: json['tripType'],
      locationLink: json['locationLink'],

    );
  }
}